python/ray/dag/compiled_dag_node.py

```diff
@@ -150,10 +169,25 @@ def __init__(self, idx: int, dag_node: "ray.dag.DAGNode"):
         """
         self.idx = idx
         self.dag_node = dag_node
+        self.arg_idx_to_tensor_meta: Dict[int, Dict[str, Any]] = {}
 
         self.downstream_node_idxs = set()
         self.output_channel = None
 
+        self.output_wrapper_fn = None
+        if self.dag_node.type_hint is not None:
+            if isinstance(self.dag_node.type_hint, TorchTensorType):
+                # Wrap outputs produced by this task to indicate that it
+                # should be specially serialized.
+                self.output_wrapper_fn = lambda t: _TorchTensorWrapper(
+                    t, self.dag_node.type_hint
+                )
```

Comment:
Should we throw an exception if type_hint is not None and is also not an instance of TorchTensorType?